consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,649 @@
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.info.call_args
+            self.assertTrue('Current fork wait duration (%f) '
+                            'less than new fork wait duration (%f)' in message)
+
+        # Subtest 2: when new fork head has the smaller wait duration
+
+        # change new_fork_mock_wait_certificate duration to a smaller value
+        new_fork_mock_wait_certificate.duration = 0.0
+
+        # set mock_utils.deserialize_wait_certificate
+        # to return a specific value for each fork_head
+        # with cur_fork_head being deserialized first
+        mock_utils.deserialize_wait_certificate.side_effect = \
+            [mock_wait_certificate,
+             new_fork_mock_wait_certificate]
+
+        # check test
+        with mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                        'LOGGER') as mock_logger:
+            fork_resolver.compare_forks(
+                cur_fork_head=mock_cur_fork_header,
+                new_fork_head=mock_new_fork_header)
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.info.call_args
+            self.assertTrue('New fork wait duration (%f) '
+                            'less than current fork wait duration '
+                            in message)
+
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.utils')
+    def test_different_previous_block_id(
+            self,
+            mock_utils,
+            mock_validator_registry_view,
+            mock_consensus_state,
+            mock_poet_enclave_factory,
+            mock_consensus_state_store,
+            mock_poet_config_view,
+            mock_block_wrapper):
+
+        """ When both current and new fork heads are valid
+            PoET blocks with different previous block ids,
+            the test verifies that the one with
+            the higher aggregate local mean wins
```

Comment:
Yes....please update the comment as you only call out what happens when one of the aggregate local means is greater than the other, but no equal.